### PR TITLE
Remove automatic -loader module name extension

### PR DESCRIPTION
**Pull request by iDams**
**RESOLVING CONFLICTS**

- [ ] Tests for the changes have been added (for bug fixes / features)

No tests have been added. As @jhnns said

> Tests should usually check for actual behavior, not for behavior that used to be there.

- [ ] Docs have been added / updated (for bug fixes / features)

Loader part on the new documentation is not written yet.

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe: 

**What is the current behavior?** (You can also link to an open issue here)
webpack automatically appends `-loader` to a module when the first module lookup failed

**What is the new behavior?**
webpack should not append `-loader`

**Does this PR introduce a breaking change?**
- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the following... 

* Impact: everyone who is using module name without `-loader` will have adjust their webpack configs
* Migration path for existing applications: always append `-loader` at the end of module name in webpack config
* Github Issue(s) this is regarding: #2986

### CODE REVIEW

File: examples/loader/example.js [hunk 1]
```diff
@@ -3,4 +3,4 @@ console.dir(require("./loader!./file"));
 
 // use buildin json loader
 console.dir(require("./test.json")); // default by extension
-console.dir(require("!json!./test.json")); // manual
\ No newline at end of file
+console.dir(require("!json!./test.json")); // manual
```
Comment (Kovensky) [inline examples/loader/example.js]: This doesn't look like `json-loader` 😉

File: examples/loader/webpack.config.js [hunk 2]
```diff
@@ -4,4 +4,4 @@ module.exports = {
 			{ test: /\.json$/, loader: "json" }
```
Comment (Kovensky) [inline examples/loader/webpack.config.js]: This also looks like it needs updating

File: examples/web-worker/webpack.config.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
 var webpack = require("../../");
 module.exports = {
-    plugins: [
```
Comment (jhnns) [inline examples/web-worker/webpack.config.js]: Seems like you've converted tabs to spaces. Please revert that change here.

**Comment by jhnns** [comment]
These changes look good. Currently, there's one failing test but this test is currently also failing on master so it's not related to this change.

Could you revert the whitespace change (as requested) and rebase the branch against master to make it up-to-date?

**Comment by iDams** [comment]
@jhnns Should be good now!

**Comment by jhnns** [comment]
@iDams awesome, thanks! 👍

**Comment by iDams** [comment]
@jhnns yay!

File: examples/web-worker/README.md [hunk 4]
```diff
@@ -358,9 +358,16 @@ Child worker:
...
+ 8ee92337ff6112b0fe42a6aa1f479e12b4423ae3
     chunk    {1} hash.worker.js (main) 162 bytes [entry] [rendered]
-        > main [1] ./worker.js 
+        > main [1] ./worker.js
         [1] ./worker.js 162 bytes {1} [built]
+=======
+    chunk    {1} hash.worker.js (main) 168 bytes [entry] [rendered]
+        > main [1] ./worker.js
+        [1] ./worker.js 168 bytes {1} [built]
+>>>>>>> fix-2986: Remove automatic -loader module name extension
```
Comment (sokra) [inline examples/web-worker/README.md]: git conflict!

Comment (iDams) [inline examples/web-worker/README.md]: Arf :/
Should I make a PR to fix it?

Comment (jhnns) [inline examples/web-worker/README.md]: Haha, damn it!
PR is ready: https://github.com/webpack/webpack/pull/3227

**Comment by kromit** [comment]
would be nice if such breaking changes would find a place on the release page :)

**Pull request merged**